polling_stations/apps/data_finder/helpers/routing.py

```diff
@@ -18,60 +17,64 @@ class RoutingHelper:
     }
 
     def __init__(self, postcode):
-        self.postcode = Postcode(postcode).without_space
-        self.get_addresses()
-        self.get_councils_from_blacklist()
+        self.postcode = Postcode(postcode)
+        self.addresses = self.get_addresses()
 
     def get_addresses(self):
-        self.addresses = ResidentialAddress.objects.filter(postcode=self.postcode)
-        return self.addresses
+        return Address.objects.uprns_for_postcode(self.postcode.with_space)
 
-    def get_councils_from_blacklist(self):
-        # if this postcode appears in the blacklist table
-        # return a list of any council ids attached to it
-        # if it is not in the table, we will return []
-        blacklist = Blacklist.objects.filter(postcode=self.postcode)
-        self.councils = [row.lad for row in blacklist]
-        return self.councils
+    @property
+    def councils(self):
+        council_ids = {a.council for a in self.addresses}
+        if len(council_ids) == 1:
+            return None
+        else:
+            return list(council_ids)
+
+    @property
+    def polling_stations(self):
+        return {address.polling_station_id for address in self.addresses}
 
     @property
     def has_addresses(self):
         return bool(self.addresses)
 
     @property
-    def has_single_address(self):
-        return self.addresses.count == 1
+    def no_stations(self):
+        """Return true if there are addresses, but no polling station information"""
+        return self.polling_stations == {""}
 
     @property
-    def address_have_single_station(self):
-        stations = self.addresses.values("polling_station_id").distinct()
-        return len(stations) == 1
+    def addresses_have_single_station(self):
+        """Check if all addresses have the same polling station id and
+        that it is not an empty string"""
+        if len(self.polling_stations) == 1:
+            return bool(min(self.polling_stations))
```

Comment:
Maybe a clearer way to write this would be `list(self.polling_stations)[0]` rather than `min(self.polling_stations)` ?